salt/config/__init__.py

```diff
@@ -4154,10 +4154,10 @@ def api_config(path):
     need to be stubbed out for salt-api
     '''
     # Let's grab a copy of salt-api's required defaults
-    opts = DEFAULT_API_OPTS
+    opts = DEFAULT_API_OPTS.copy()
 
     # Let's override them with salt's master opts
-    opts.update(client_config(path, defaults=DEFAULT_MASTER_OPTS))
+    opts.update(client_config(path, defaults=DEFAULT_MASTER_OPTS.copy()))
```

Comment:
Are we requiring a `.copy` in all the cases, or just some?

If it's all the cases, it might be useful to just change to a pattern like [this one](https://stackoverflow.com/a/39440252/344286), which makes it no longer possible for someone to accidentally forget `.copy()`. Thinking about it, one wouldn't even need to use frozenset, something like this would work just fine:

```
DEFAULT_API_OPTS = tuple({'foo': 'bar', 'salt': 'sodium chloride', 'water': 'dihydrogen monoxide'}.items())
...

opts.update(..., defaults=dict(DEFAULT_API_OPTS))
```

Not sure if the change is worth it, but it would at least enforce copying!

Reply:
https://github.com/saltstack/salt/pull/51245